sympy/core/sympify.py

```diff
@@ -505,7 +505,15 @@ def _sympify(a):
     see: sympify
 
     """
-    return sympify(a, strict=True)
+    symp = sympify(a, strict=True)
+    if expected_type is None:
+        return symp
+    if not isinstance(expected_type, list):
+        raise TypeError('Expected Type must  be list')
+    for i in expected_type:
+        if not isinstance(symp, i):
```

Comment:
I don't see why this needs to be a list. We can use both a tuple of types or a single type with `isinstance`:
```julia
In [1]: isinstance(1, int)                                                                                                                                    
Out[1]: True

In [2]: isinstance(1, (int, float))                                                                                                                           
Out[2]: True
```
It's also more efficient to pass the tuple to `isinstance` than to loop over the list of types calling `isinstance` repeatedly:
```julia
In [6]: t = (float, int)                                                                                                                                      

In [7]: %timeit isinstance(1, t)                                                                                                                              
240 ns ± 2.29 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [8]: %timeit for ti in t: isinstance(1, ti)                                                                                                                
444 ns ± 45.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```